Source-Code/FlareDraw.cs

```diff
@@ -192,9 +192,22 @@ private void GenerateBodyFlares()
             //ListChildren(sm.systemPrefab.rootBody, 0);
             //--- HACK--
 
+            //If Kerbin is parented to the Sun, set its SMA- otherwise iterate
+            //through celestial bodies to locate which is parented to the Sun
+            //and has Kerbin as a child. Set the highest parents SMA to kerbinSMA
             if(BodyFlare.kerbinSMA <= 0.0)
             {
-                BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                if(FlightGlobals.Bodies[1].referenceBody == FlightGlobals.Bodies[0])
+                    BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                else
+                {                    
+                    foreach(CelestialBody current in FlightGlobals.Bodies)
```

Comment:
I've manually integrated your changes, since GitHub insists it can't automatically merge it, and I don't have the patience to deal with the GitHub merge process this morning (worst process I've used in many *mumblemumble* years of software engineering), so I'm closing this pull request.  I added a test after the foreach loop to throw an exception if kerbinSMA is still uninitialized, and I added some curly-braces.  Thanks again for sorting this problem out.